src/Tools/dotnet-openapi/src/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,208 @@
+        public const string OpenApiReference = "OpenApiReference";
+        public const string OpenApiProjectReference = "OpenApiProjectReference";
+        protected const string SourceUrlAttrName = "SourceUrl";
+        
+        public BaseCommand(CommandLineApplication parent, string name)
+        {
+            Parent = parent;
+            Name = name;
+            Out = parent.Out ?? Out;
+            Error = parent.Error ?? Error;
+
+            SourceFileArg = Argument(SourceFileArgName, $"The openapi file to {name}. This can be a path to a local openapi file, " +
+               "a URI to a remote openapi file or a path to a *.csproj file containing openapi endpoints", multipleValues: true);
+            ProjectFileOption = Option("-p|--project", "The project to add a reference to", CommandOptionType.SingleValue);
+
+            Help = HelpOption("-?|-h|--help");
+            if(Parent is Application)
+            {
+                WorkingDir = ((Application)Parent).WorkingDir;
+            }
+            else
+            {
+                WorkingDir = ((Application)Parent.Parent).WorkingDir;
+            }
+
+            OnExecute(ExecuteAsync);
+        }
+
+        public CommandOption ProjectFileOption { get; }
+
+        public CommandArgument SourceFileArg { get; }
+         
+        internal CommandOption Help { get; }
+
+        protected abstract Task<int> ExecuteCoreAsync();
+
+        protected virtual bool ValidateArguments()
+        {
+            Ensure.NotNullOrEmpty(SourceFileArg.Value, SourceFileArgName);
+            return true;
+        }
+
+        private async Task<int> ExecuteAsync()
+        {
+            if (!ValidateArguments() || Help.HasValue())
+            {
+                ShowHelp();
+                return 1;
+            }
+
+            return await ExecuteCoreAsync();
+        }
+
+        internal FileInfo ResolveProjectFile(CommandOption projectOption)
+        {
+            string csproj;
+            if (projectOption.HasValue())
+            {
+                csproj = projectOption.Value();
+                csproj = Path.Combine(WorkingDir, csproj);
+                if (!File.Exists(csproj))
+                {
+                    Error.Write("The given csproj does not exist.");
+                }
+            }
+            else
+            {
+                var csprojs = Directory.GetFiles(WorkingDir, "*.csproj", SearchOption.TopDirectoryOnly);
+                if (csprojs.Length == 0)
+                {
+                    Error.Write("No csproj files were found in the current directory. Either move to a new directory or provide the project explicitly");
+                }
+                if (csprojs.Length > 1)
+                {
+                    Error.Write("More than one csproj was found in this directory, either remove a duplicate or explicitly provide the project.");
+                }
+
+                csproj = csprojs.Single();
```

Comment:
```suggestion
                csproj = csprojs[0];
```
No need for Linq here